spec/controllers/comments_controller_spec.rb

```diff
@@ -2114,15 +2114,18 @@
   end
 
   describe "GET #index" do
-    it "errors when not logged in as admin" do
+    it "redirects to 404 when not logged in as admin" do
       get :index
-      expect(flash[:error]).to eq "Sorry, you don't have permission to access that page."
+
+      expect(response.status).to eq(404)
```

Comment:
I thought about the test failures a bit and they do make sense. We're sort of simulating a proper 404 response with the code we have: you go to the comment index and you get redirected to the 404 page -- ergo a 302 response. So changing this to `it_redirects_to_simple("/404")` like we do with hit_count_controller_spec.rb will probably clear up the test failures just fine.

(If you're still having issues with your dev environment, it's fine to edit the file here on GitHub and rely on GitHub Actions to see if the change works. We've definitely all done that once or twice. 😆)

Reply:
thanks @sarken !! i'll give this a shot. Still having some issues with the ES portion of my local, but I think the initial issue at least was that I wasn't using the correct version, so I'm (hopefully) slowly getting there!

Reply:
Are you by any chance on a Mac with the new M1 chip? I ask because the Archive's version of Elasticsearch is incompatible with the new Macs. 😩  

Fortunately, unless something catastrophic happens, we'll be updating Elasticsearch in the next week or two -- even if you're not using a new Mac, that might help.

Reply:
I'm using a new mac, but not one with an M1 chip! 🙀 

Is there an issue/ticket I can follow along with so I know when ES has been updated? I've been trying various things but having the updated version might just solve it (fingers crossed!). For a while I thought it was because something funky happened when I tried to downgrade the ES version, but that doesn't quite seem to be it sadly

Reply:
>I'm using a new mac, but not one with an M1 chip!

Huh, weird! I have it running on Big Sur on a Mac that's a couple years old, but I was using a different OS version and a different Homebrew version, so maybe things have changed since then. 

However, since you're not using the new chip, you might have some luck with [Docker](https://github.com/otwcode/otwarchive/wiki/Docker). That also doesn't work quite right with the new chip yet, so I'm not using it, but some of our other volunteers are, so if you decide to try it and need help, you can email the otw-coders@transformativeworks.org mailing list.

>Is there an issue/ticket I can follow along with so I know when ES has been updated?

There sure is! The pull request is #3570 and the Jira issue is [AO3-5671](https://otwarchive.atlassian.net/browse/AO3-5671).

Also, thank you for your kind words below! I'm very glad to hear you enjoyed the learning process and I hope we get to see more from you in the future!

Reply:
that's awesome, thank you so much! I'll keep tabs on the PR, and if that really doesn't solve my issue I'll give docker a try! I definitely hope to contribute more, I've scoped some more issues out so hopefully I'll be able to work on one soon :D

And you're welcome! Gotta give credit where credit is due 🤓

Reply:
Hi again, @gardengal! I just got Elasticsearch working thanks to @elzj and wanted to let you know what worked for me in case it also helps you:

* If you're using Elasticsearch and/or Java/openjdk with Homebrew, uninstall them
* [Download Elasticsearch 6.8.18 from the Elasticsearch website](https://www.elastic.co/downloads/past-releases/elasticsearch-6-8-18), unzip it, and drag the whole elasticsearch-6.8.0 folder somewhere easy to find, e.g. your Documents folder
* Download Java from [java.com](https://java.com) (this should be version 8) and install it
* If you haven't already, [disable Gatekeeper](https://disable-gatekeeper.github.io/) -- until I did that, I ran into [errors like these](https://github.com/elastic/elasticsearch/issues/49664) when starting Elasticsearch
* `cd Documents/elasticsearch-6.8.0/bin` (or wherever) and then `./elasticsearch` to start Elasticsearch -- you can [set $JAVA_HOME](https://mkyong.com/java/how-to-set-java_home-environment-variable-on-mac-os-x/) if it yells about that
* Open a new tab in Terminal and start your Resque workers: `RAILS_ENV=development QUEUE=* bundle exec rake environment resque:work` 
* [Make sure you have the right port in your local.yml file and (in yet another new Terminal tab) run the indexing tasks](https://github.com/otwcode/otwarchive/wiki/OS-X-Setup#12-create-elasticsearch-indexes)

Reply:
THANK YOU once again @sarken !!! i appreciate all the help debugging this!! This seems to have worked for me. (I haven't gone back to try the original instructions, but I think some of my issues might've been stemming from my terminal not having full disk access 😅 only realized after I kept trying to run `ls` and it was saying i couldn't 😂 )

But ES is working!! i was able to index things and search!! I can't say how excited I am :D

Reply:
Woohoo! I'm glad you got it working. If you run into any more issues, feel free to email the otw-coders@transforamtiveworks.org mailing list and we'll try our best to help sort those out, too!